### ISSUE TITLE
When using grouping stderr and stdout output is no longer in order

### ISSUE THREAD

**Issue by ssbarnea** [go-task/task#779]
It took me some time to discover why the output of some commands was very confusing as the order of the console lines were not in order.

That is happening because when run using grouping both stdout and stderr are captures separated and printed separated, one after another. As you can imagine that can cause very confusing console output for script. Easy to see especially when running bash scripts with `set -x`.

It would be preferable if the subprocess would use line-buffering and print lines in order they were generated, making read of output more human friendly.

**Comment by ghostsquad** [comment]
Do you have a recommended solution for this?

**Comment by ssbarnea** [comment]
I did not look at the code but I did implemented in python something similar to tee, for very similar reasons.

The current behavior is quite bad, I observed at https://github.com/ansible/ansible-language-server/runs/6947064789?check_suite_focus=true where it took me a large amount of scrolling (several screens) to spot the real error.

Do we really need to capture stdout and stderr separately or we can just capture them combined? If we do not need to save their separated output, having a fix should be easy.
